From 1d881dd6925a5b1209fcc3da75d1f659239d6f00 Mon Sep 17 00:00:00 2001
From: Piyush Gupta <gpiyush@vmware.com>
Date: Mon, 13 Dec 2021 12:55:45 +0000
Subject: [PATCH] Fixed formatting and testing module errors.

metrics/heapster_test.go: Replaced Fatalf with Errorf and return
since Fatalf cannot be called from a go subroutine.

common/kafka/glogadapter.go: Added args as sliced parameter since
it can be of variable length.

events/sinks/riemann/driver.go: metrics/sinks/riemann/driver.go:
metrics/sources/kubelet/kubelet.go: Added formatters to Warning
and info msgs as it was missing.
---
 common/kafka/glogadapter.go            | 8 ++++----
 events/sinks/riemann/driver.go         | 4 ++--
 metrics/heapster_test.go               | 8 ++++++--
 metrics/sinks/opentsdb/driver_test.go  | 2 +-
 metrics/sinks/riemann/driver.go        | 4 ++--
 metrics/sinks/wavefront/driver_test.go | 2 +-
 metrics/sources/kubelet/kubelet.go     | 2 +-
 7 files changed, 17 insertions(+), 13 deletions(-)

diff --git a/common/kafka/glogadapter.go b/common/kafka/glogadapter.go
index b044d9c..5336e28 100644
--- a/common/kafka/glogadapter.go
+++ b/common/kafka/glogadapter.go
@@ -26,17 +26,17 @@ type GologAdapterLogger struct {
 }
 
 func (GologAdapterLogger) Debug(msg string, args ...interface{}) {
-	glog.V(6).Infof(msg, args)
+	glog.V(6).Infof(msg, args...)
 }
 
 func (GologAdapterLogger) Info(msg string, args ...interface{}) {
-	glog.Infof(msg, args)
+	glog.Infof(msg, args...)
 }
 
 func (GologAdapterLogger) Warn(msg string, args ...interface{}) {
-	glog.Warningf(msg, args)
+	glog.Warningf(msg, args...)
 }
 
 func (GologAdapterLogger) Error(msg string, args ...interface{}) {
-	glog.Errorf(msg, args)
+	glog.Errorf(msg, args...)
 }
diff --git a/events/sinks/riemann/driver.go b/events/sinks/riemann/driver.go
index 0d7ab90..7fe8461 100644
--- a/events/sinks/riemann/driver.go
+++ b/events/sinks/riemann/driver.go
@@ -103,7 +103,7 @@ func appendEvent(events []riemanngo.Event, sink *RiemannSink, event *kube_api.Ev
 	if len(events) >= sink.config.BatchSize {
 		err := riemannCommon.SendData(sink.client, events)
 		if err != nil {
-			glog.Warningf("Error sending events to Riemann: ", err)
+			glog.Warningf("Error sending events to Riemann: %v", err)
 			// client will reconnect later
 			sink.client = nil
 		}
@@ -137,7 +137,7 @@ func (sink *RiemannSink) ExportEvents(eventBatch *core.EventBatch) {
 	if len(events) > 0 {
 		err := riemannCommon.SendData(sink.client, events)
 		if err != nil {
-			glog.Warningf("Error sending events to Riemann: ", err)
+			glog.Warningf("Error sending events to Riemann: %v", err)
 			// client will reconnect later
 			sink.client = nil
 		}
diff --git a/metrics/heapster_test.go b/metrics/heapster_test.go
index 1f3b17e..73d6ca8 100644
--- a/metrics/heapster_test.go
+++ b/metrics/heapster_test.go
@@ -58,10 +58,14 @@ func TestRunSecureServer(t *testing.T) {
 
 		server, err := app.NewHeapsterApiServer(opt, emptyMetricSink, emptyNodeLister, emptyPodLister)
 		if err != nil {
-			t.Fatalf("Could not create the API server: %v", err)
+			//Replaced Fatalf with Errorf and return since it cannot be called from go subroutine.
+			t.Errorf("Could not create the API server: %v", err)
+			return
 		}
 		if err := server.RunServer(); err != nil {
-			t.Fatalf("Error in bringing up the server: %v", err)
+			//Replaced Fatalf with Errorf and return since it cannot be called from go subroutine.
+			t.Errorf("Error in bringing up the server: %v", err)
+			return
 		}
 	}()
 	if err := waitForApiserverUp(serverIP); err != nil {
diff --git a/metrics/sinks/opentsdb/driver_test.go b/metrics/sinks/opentsdb/driver_test.go
index 5d270aa..0d6497b 100644
--- a/metrics/sinks/opentsdb/driver_test.go
+++ b/metrics/sinks/opentsdb/driver_test.go
@@ -32,7 +32,7 @@ var (
 	fakePodUid       = "redis-test-uid"
 	fakeClusterName  = "fakeClusterName"
 	fakeLabel        = map[string]string{
-		"name": "redis",
+		"name":                   "redis",
 		"io.kubernetes.pod.name": "default/redis-test",
 		"pod_id":                 fakePodUid,
 		"pod_name":               fakePodName,
diff --git a/metrics/sinks/riemann/driver.go b/metrics/sinks/riemann/driver.go
index 70d81d4..523030e 100644
--- a/metrics/sinks/riemann/driver.go
+++ b/metrics/sinks/riemann/driver.go
@@ -75,7 +75,7 @@ func appendEvent(events []riemanngo.Event, sink *RiemannSink, host, name string,
 	if len(events) >= sink.config.BatchSize {
 		err := riemannCommon.SendData(sink.client, events)
 		if err != nil {
-			glog.Warningf("Error sending events to Riemann: ", err)
+			glog.Warningf("Error sending events to Riemann: %v", err)
 			// client will reconnect later
 			sink.client = nil
 		}
@@ -129,7 +129,7 @@ func (sink *RiemannSink) ExportData(dataBatch *core.DataBatch) {
 	if len(events) > 0 {
 		err := riemannCommon.SendData(sink.client, events)
 		if err != nil {
-			glog.Warningf("Error sending events to Riemann: ", err)
+			glog.Warningf("Error sending events to Riemann: %v", err)
 			// client will reconnect later
 			sink.client = nil
 		}
diff --git a/metrics/sinks/wavefront/driver_test.go b/metrics/sinks/wavefront/driver_test.go
index d1ef986..708e00e 100644
--- a/metrics/sinks/wavefront/driver_test.go
+++ b/metrics/sinks/wavefront/driver_test.go
@@ -29,7 +29,7 @@ var (
 	fakePodName = "redis-test"
 	fakePodUid  = "redis-test-uid"
 	fakeLabel   = map[string]string{
-		"name": "redis",
+		"name":                   "redis",
 		"io.kubernetes.pod.name": "default/redis-test",
 		"pod_id":                 fakePodUid,
 		"namespace_name":         "default",
diff --git a/metrics/sources/kubelet/kubelet.go b/metrics/sources/kubelet/kubelet.go
index 1bfe947..5386e97 100644
--- a/metrics/sources/kubelet/kubelet.go
+++ b/metrics/sources/kubelet/kubelet.go
@@ -223,7 +223,7 @@ metricloop:
 			mv.ValueType = ValueFloat
 			mv.FloatValue = newest.FloatValue
 		default:
-			glog.V(4).Infof("Skipping %s: unknown custom metric format", spec.Name, spec.Format)
+			glog.V(4).Infof("Skipping %s: unknown custom metric format %v", spec.Name, spec.Format)
 			continue metricloop
 		}
 
-- 
2.23.3

